File: common/src/main/java/com/ssblur/scriptor/item/bound/BoundTool.java
package com.ssblur.scriptor.item.bound;

import com.ssblur.scriptor.color.CustomColors;
import dev.architectury.platform.Platform;
import dev.architectury.registry.client.rendering.ColorHandlerRegistry;
import net.fabricmc.api.EnvType;
import net.minecraft.tags.TagKey;
import net.minecraft.world.item.DiggerItem;
import net.minecraft.world.item.Tier;
import net.minecraft.world.level.block.Block;

public class BoundTool extends DiggerItem {
  TagKey<Block> tagKey;
  public BoundTool(Tier tier, TagKey<Block> tagKey, Properties properties) {
    super(tier, tagKey, properties);
    this.tagKey = tagKey;

    if(Platform.getEnv() == EnvType.CLIENT)
      ColorHandlerRegistry.registerItemColors((itemStack, t) -> t == 1 ? CustomColors.INSTANCE.getColor(itemStack) : 0xFFFFFFFF, this);
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/item/tools/bound/BoundTool.kt
package com.ssblur.scriptor.item.tools.bound

import com.ssblur.scriptor.data.components.ScriptorDataComponents
import net.minecraft.ChatFormatting
import net.minecraft.network.chat.Component
import net.minecraft.tags.TagKey
import net.minecraft.world.item.DiggerItem
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Tier
import net.minecraft.world.item.TooltipFlag
import net.minecraft.world.level.block.Block

class BoundTool(tier: Tier, tagKey: TagKey<Block>, properties: Properties):
  DiggerItem(tier, tagKey, properties) {
  override fun appendHoverText(
    itemStack: ItemStack,
    tooltipContext: TooltipContext,
    list: MutableList<Component>,
    tooltipFlag: TooltipFlag
  ) {
    super.appendHoverText(itemStack, tooltipContext, list, tooltipFlag)
    if(!itemStack.has(ScriptorDataComponents.EXPIRES))
      list.add(Component.translatable("lore.scriptor.bound_tool").withStyle(ChatFormatting.GRAY))
  }
}
